src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -95,17 +100,23 @@ TryDropMarkedShards(bool waitForCleanupLock)
  * This is to ensure that this function is not being run concurrently.
  * Otherwise really bad race conditions are possible, such as removing all
  * placements of a shard. waitForCleanupLock indicates if this function should
- * wait for this lock or returns with a warning.
+ * wait for this lock or error out.
+ *
+ * The function returns true if all shards were successfuly dropped and fills
+ * removedShardCount with the number of shards that were dropped.
  */
-static int
-DropMarkedShards(bool waitForCleanupLock)
+bool
+DropMarkedShards(bool waitForCleanupLock, int *removedShardCount)
```

Comment:
do we need this changes if we remove the retry loop?

Reply:
I think the reason for this change was that when `DropMarkedShards` error we don't want to return just "-1" to the callers but the actual number of shards that were dropped. But with the current version I don't think `DropMarkedShards` can error this number incorrectly because we have `TryDropShard` and it catches errors already. Therefore we can go back to the previous version